### DISCUSSION TITLE
Take more time to decrypt

### DISCUSSION THREAD

**Discussion by kamalupasena** [krzyzanowskim/CryptoSwift#962] [General]
Hi i am currently using this library for file decryption and its work fine but the issue i am facing was its take more time to decrypt. my file size around 10 MB and it take around 5mins to decrypt to work. this is my decryption  code 
   
`let decryptedPart = try! AES(key: keyArray, iv: ivDecodeData, blockMode: .CTR, padding: NoPadding()).decrypt(encrptedPart)`

i am working in swift 2.2 and can you give me a solution for this issue ?

**Comment by krzyzanowskim** [comment]
Upgrade to Swift 3 and use Release build with "whole module optimization" enabled. According to my latest tests it should take ~4s to process 10MB when run on iPhone6 hardware.

**Comment by kamalupasena** [comment]
i am currently using 
`pod 'CryptoSwift' , '~>0.5.2'`

which version should i use ?

**Comment by krzyzanowskim** [comment]
0.6.5 or above

**Comment by kamalupasena** [comment]
is there a way to do it without updating it into swift 3.0 ?

**Comment by krzyzanowskim** [comment]
For the best performance, you'll need the latest release, Swift 3 compatible. I have no plans to maintain Swift 2.2.

**Comment by kamalupasena** [comment]
Hi
I have update my code to swift 3 and do same decryption but it still take about 5 min to do the decryption can you tell me what i am doing wrong here ?

**Comment by krzyzanowskim** [comment]
share you sample project please, to verify

**Comment by kamalupasena** [comment]
i am using  
`pod 'CryptoSwift'`

```
let ivDecodeData = Data(base64Encoded: myChapter.iv!, options: [])?.bytes

        let keySha = myChapter.secret!.data(using: String.Encoding.utf8)!.sha1().bytes


        var keyArray = Array<UInt8>()

        for i in 0...15{
            keyArray.append(keySha[i])
        }

        print("Start Time ", NSDate().timeIntervalSince1970)

        let decryptedPart = try! AES(key: keyArray, iv: ivDecodeData, blockMode: .CTR, padding: NoPadding()).decrypt(encrptedPart)

         print("Finished Time ", NSDate().timeIntervalSince1970)

        var decryptedPartSemi = [UInt8]()
```

ill share the sample project with you but for the time being this is my code snippet

**Comment by krzyzanowskim** [comment]
no way. I took 7.5 s. on my iPhone 6. You must build it in "debug" mode.

**Comment by kamalupasena** [comment]
i have created  a sample application 

[Here](https://drive.google.com/file/d/0B33otd55T2ObZnBpU2JEdW1mRGs/view?usp=sharing)
and i am testing this on my iPhone 5

**Comment by kamalupasena** [comment]
Did you able to run the project that i have share with you ?

**Comment by krzyzanowskim** [comment]
yes, here the results

iPhone6

```
File Downloading at 100 %
Downloaded file successfully
Decryption Start Time  1478206541.37543
Decryption Finished Time  1478206548.36963
true
```

iPad Mini (iPhone 4s)

```
File Downloading at 100 %
Downloaded file successfully
Decryption Start Time  1478213743.40164
Decryption Finished Time  1478213762.92804
true
```

**Comment by kamalupasena** [comment]
So what i am doing wrong there ? because i'am getting following results 

`Start Time  1478213260.39949
Finished Time  1478213607.22812`

**Comment by krzyzanowskim** [comment]
I'm pretty sure you're testing Debug builds, not Release builds.

**Comment by kamalupasena** [comment]
Can you tell me how to change it ?

**Comment by krzyzanowskim** [comment]
sure. Like this:

![screen shot 2016-11-04 at 00 16 44](https://cloud.githubusercontent.com/assets/758033/19989022/02aa8a24-a224-11e6-87bb-3c6d4a2fd5be.png)

Also... you may want update CryptoSwift target in Pods (as seen in screenshot) to enable "Whole Module Optimization"

**Comment by kamalupasena** [comment]
Thank you very much, changing it to Release and enable Whole Module Optimization cut the decryption time in to this 

`Start Time  1478215560.89765
Finished Time  1478215574.24821`  

Thank you very much again for your support and this awesome library
